paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -556,41 +707,49 @@ internal class IntentConfirmationHandler(
     /**
      * Types of errors that can occur when confirming an intent.
      */
-    enum class ErrorType {
+    sealed interface ErrorType {
         /**
          * Fatal confirmation error that occurred while confirming a payment. This should never happen.
          */
-        Fatal,
+        data object Fatal : ErrorType
 
         /**
          * Indicates an error when processing a payment during the confirmation process.
          */
-        Payment,
+        data object Payment : ErrorType
 
         /**
          * Indicates an internal process error occurred during the confirmation process.
          */
-        Internal,
+        data object Internal : ErrorType
 
         /**
          * Indicates an error occurred when confirming with external payment methods
          */
-        ExternalPaymentMethod,
+        data object ExternalPaymentMethod : ErrorType
+
+        /**
+         * Indicates an error occurred when confirming with Google Pay
+         */
+        data class GooglePay(val errorCode: Int) : ErrorType
     }
```

Comment:
Eventually, we should keep these error types internal and do payment status reporting inside the handler itself